### PR TITLE
Add assets models.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

File: src/ostorlab/runtimes/local/models/models.py [hunk 1]
```diff
@@ -483,3 +493,206 @@ def is_valid(api_key: str) -> bool:
...
+
+
+class Asset(Base):
+    __tablename__ = "asset"
+    id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+    type = sqlalchemy.Column(sqlalchemy.String(50))
+    __mapper_args__ = {
+        "polymorphic_identity": "asset",
+        "polymorphic_on": type,
+    }
```
Comment (3asm) [inline src/ostorlab/runtimes/local/models/models.py:505]: Nice work.

**Review by 3asm — APPROVED** [review]

**Review by mohsinenar — APPROVED** [review]

**Review by PiranhaSa — APPROVED** [review]
lgtm

File: src/ostorlab/runtimes/local/models/models.py [hunk 2]
```diff
@@ -483,3 +493,206 @@ def is_valid(api_key: str) -> bool:
...
+            session.commit()
+            return asset
+
+
+class Url(Asset):
+    __tablename__ = "urls"
+    id = sqlalchemy.Column(
+        sqlalchemy.Integer, sqlalchemy.ForeignKey("asset.id"), primary_key=True
+    )
+    links = sqlalchemy.Column(sqlalchemy.String(1024))
```
Comment (oussamaessaji) [inline src/ostorlab/runtimes/local/models/models.py:646]: why not array https://docs.sqlalchemy.org/en/20/core/type_basics.html#sqlalchemy.types.ARRAY

File: src/ostorlab/runtimes/local/models/models.py [hunk 3]
```diff
@@ -483,3 +493,206 @@ def is_valid(api_key: str) -> bool:
...
+    __mapper_args__ = {
+        "polymorphic_identity": "android_file",
+    }
+
+    @staticmethod
+    def create(package_name: str = "", path: str = ""):
+        """Persist the android file information in the database.
+
+        Args:
+            bundle_id: The application identifier.
```
Comment (benyissa) [inline src/ostorlab/runtimes/local/models/models.py:526]:
```suggestion
            package_name: The application identifier.
```

**Review by benyissa — CHANGES_REQUESTED** [review]
```python
class Network(Asset):
    __tablename__ = "network"
    id = sqlalchemy.Column(
        sqlalchemy.Integer, sqlalchemy.ForeignKey("asset.id"), primary_key=True
    )
    cidr = sqlalchemy.Column(sqlalchemy.String(50))
    __mapper_args__ = {
        "polymorphic_identity": "network",
    }
    
    ...
    
        @staticmethod
    def create(cidr: str):
       ....
        with Database() as session:
            network = Network(
                cidr=cidr,
            )
            session.add(network)
            session.commit()
            return network
```

### Benefit:
**Storage Efficiency**: Using CIDR notation (`cidr` field) means storing one entry per network, regardless of the number of IPs. Avoiding the storage of large lists of IPs, especially for large networks.

**Review by elyousfi5 — CHANGES_REQUESTED** [review]

File: src/ostorlab/runtimes/local/models/models.py (hunk 2 above)
Comment (elyousfi5) [inline src/ostorlab/runtimes/local/models/models.py:646]: why string ? i know ARRAY is not supported in sqlite but you can use json i think its better

File: src/ostorlab/runtimes/local/models/models.py [hunk 4]
```diff
@@ -483,3 +493,206 @@ def is_valid(api_key: str) -> bool:
...
+            session.commit()
+            return asset
+
+
+class Network(Asset):
+    __tablename__ = "network"
+    id = sqlalchemy.Column(
+        sqlalchemy.Integer, sqlalchemy.ForeignKey("asset.id"), primary_key=True
+    )
+    ips = sqlalchemy.Column(sqlalchemy.String(1024))
```
Comment (elyousfi5) [inline src/ostorlab/runtimes/local/models/models.py:676]: same !

File: tests/runtimes/local/models/models_test.py [hunk 5]
```diff
@@ -262,3 +265,145 @@ def testModelsAPIKeyValidation_whenKeyIsInvalid_returnsFalse(
...
+        ips = json.loads(session.query(models.Network).all()[0].ips)
+        assert ips == ["8.8.8.8", "42.42.42.42"]
+
+
+def testAssetModels_whenCreateUrl_assetCreated(
+    mocker: plugin.MockerFixture, db_engine_path: str
+) -> None:
+    """Ensure we correctly persist the list of target URLs information."""
+    mocker.patch.object(models, "ENGINE_URL", db_engine_path)
+    models.Url.create(links=["https://example.com", "https://example42.com"])
```
Comment (elyousfi5) [inline tests/runtimes/local/models/models_test.py:288]: i think we need method here.

image

or they will be considered as domain names ?

File: tests/serve_app/oxo_test.py [hunk 6]
```diff
@@ -44,7 +44,7 @@ def testImportScanMutation_always_shouldImportScan(
             "/graphql", data=data, content_type="multipart/form-data"
         )
 
-        assert response.status_code == 200, response.content
+        assert response.status_code == 200, response.get_json()
```
[the remark this answers is not in the document]
Comment (deadly-panda) [reply tests/serve_app/oxo_test.py:47]: I'm not sure I understand, for this case it's json, sinon we use `response.text` ?

**Comment by 3asm** [comment]
Ping @deadly-panda on PR status.

File: src/ostorlab/runtimes/local/models/models.py [hunk 7]
```diff
@@ -534,3 +544,206 @@ def refresh() -> "APIKey":
...
+    )
+    package_name = sqlalchemy.Column(sqlalchemy.String(255))
+    path = sqlalchemy.Column(sqlalchemy.String(1024))
+
+    __mapper_args__ = {
+        "polymorphic_identity": "android_file",
+    }
+
+    @staticmethod
+    def create(package_name: str = "", path: str = ""):
```
Comment (benyissa) [inline src/ostorlab/runtimes/local/models/models.py:573]: missing return type.

**Review by 3asm — APPROVED** [review]

**Pull request merged**